### ISSUE TITLE
CKAN errors out when attempting to load or update to Cetera's Suit Pack for TRR v 1.03 on KSP 1.3.1

### ISSUE THREAD

**Issue by Murdabenne** [KSP-CKAN/CKAN#2198]
CKAN errors out when attempting to load or update to Cetera's Suit Pack for TRR v 1.03 on KSP 1.3.1

"_Trying to install Ceteras-Suit-Pack 1.03, but it's not downloaded or download is corrupted_"

Checking the download directory for CKAN, the file is downloaded, it is in the download directory "3395426C-Ceteras-Suit-Pack-1.03.zip", and it manually unzips just fine.  So the issue appears to be either with CKAN or the metadata in netkan re:checksum. 

**CKAN Version:** 
1.22.6
**Operating System:** 
Win10-64 Pro
**The issue you are experiencing:** 
Checksumming logic error (possibly)
**How to recreate this issue:** 
attempt to add Cetera's Suit Pack 1.03
**CKAN error codes (if applicable):** 
```
CKAN.FileNotFoundKraken: Trying to install Ceteras-Suit-Pack 1.03, but it's not downloaded or download is corrupted
   at CKAN.ModuleInstaller.Install(CkanModule module, String filename)
   at CKAN.ModuleInstaller.InstallList(ModuleResolution modules, RelationshipResolverOptions options)
   at CKAN.Main.WasSuccessful(Action action)
   at CKAN.Main.InstallMods(Object sender, DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)

```

**Comment by HebaruSan** [comment]
Finally managed to get some useful output; SharpZipLib doesn't make it easy to get info about what failed. This is internal to `ZipFile.TestArchive`.

"Compression method not supported":

```
10048 [1] ERROR CKAN.NetFileCache (null) - Exception during test - 'Compression method not supported'
10056 [1] ERROR CKAN.NetFileCache (null) - ZIP invalid: EntryHeader, Gamedata/TRR-Cetera/CeteraIcons.cfg, 1, 0, False
10057 [1] ERROR CKAN.NetFileCache (null) - ZIP invalid: EntryComplete, Gamedata/TRR-Cetera/CeteraIcons.cfg, 1, 0, False
10057 [1] ERROR CKAN.NetFileCache (null) - ZIP invalid: MiscellaneousTests, Gamedata/TRR-Cetera/CeteraIcons.cfg, 1, 0, False
```

That file looks fine:

```
 Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
--------  ------  ------- ---- ---------- ----- --------  ----
    2530  Def64N      814  68% 2017-11-23 09:37 b313c435  Gamedata/TRR-Cetera/CeteraIcons.cfg
```

Our SharpZipLib seems to be the latest version, so we can't hope to gain support for more compression methods by updating. If we can't figure it out, we may need to ask Cetera how this zip was created.

**Comment by HebaruSan** [comment]
Forum PM from Cetera:

> I used 7-zip.  I think I see the issue, though.  I was preparing some stuff to be archived on my NAS, moved, etc.  I didn't change my settings back to default in 7-zip after doing that, so this most recent zip was set to compression level Ultra, using Deflate64, dictionary size of 64kb, word size 128.  I think the main difference is the Deflate64 and dictionary size of 64kb.  I believe standard zips use Deflate, with 32kb.

And indeed, SharpLibZip [says](https://github.com/icsharpcode/SharpZipLib/wiki):

> Zip64 is supported while Deflate64 is not yet supported.

And they might not want to support it, because it's proprietary? https://github.com/icsharpcode/SharpZipLib/issues/23

**Comment by HebaruSan** [comment]
Cetera has released 1.04, which seems not to have this problem.

Since the cause was an unsupported file format handled by a third party library, I don't think the CKAN team is going to try to address this. Closing as the problem no longer happens with the latest version of the mod.
